app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -166,12 +166,16 @@ class UndertakingController @Inject() (
   }
 
   def postConfirmEmail: Action[AnyContent] = enrolled.async { implicit request =>
-    handleConfirmEmailPost[UndertakingJourney](
-      previous = routes.UndertakingController.getConfirmEmail(),
-      next = routes.UndertakingController.getAddBusiness(),
-      formAction = routes.UndertakingController.postConfirmEmail(),
-      generateVerifyEmailUrl = (id: String) => routes.UndertakingController.getVerifyEmail(id).url
-    )
+    implicit val eori: EORI = request.eoriNumber
```

Comment:
`handleConfirmEmailPost` is intended to handle the same submission for both the undertaking and business lead promote flows. This code looks like it might generate a warning on the unhandled None case.

It's probably better to have the shared code take a type parameter and do the journey check for you and the shared code provides a `withJourneyOrRedirect` helper method which should probably be used by `handleConfirmEmailPost` to check that the journey exists.